ISM43362Interface.cpp

```diff
@@ -454,7 +454,7 @@ int ISM43362Interface::socket_recv(void *handle, void *data, unsigned size)
         } else if (read_amount < 0) {
             socket->connected = false;
             _mutex.unlock();
-            return NSAPI_ERROR_CONNECTION_LOST;
+            return 0;
         }
```

Comment:
Same here. Question: socket_recv is supposed to return
     *  @return         Number of received bytes on success, negative error
     *                  code on failure
here, if the socket is not valid, shouldn't it be considered a failure ?